### DISCUSSION TITLE
03_pytorch_computer_vision FashionMNIST

### DISCUSSION THREAD

**Discussion by kydaly** [mrdbourke/pytorch-deep-learning#1032] [Q&A]
I had a question regarding video 96 at time 14:48:38. When he uses .classes and .class_to_idx how can I look up what attributes the class has? I know the dropdown menu shows different attributes associated with the class but the descriptions are very vague. The FashionMNIST documentation does not provide the attribute information either. I appreciate any further details anyone can provide.
Thanks!

**Comment by mrdbourke** [comment]
Hi @kydaly ,

You're right, sometimes the PyTorch documentation can be a little lacking.

However, you can often find tidbits about things by reading the source code:

<img width="898" alt="Screenshot 2024-08-22 at 8 47 02 PM" src="https://github.com/user-attachments/assets/33c1db8b-a7aa-486c-92f0-b715c32f711c">

See here: https://pytorch.org/vision/stable/_modules/torchvision/datasets/mnist.html#FashionMNIST

<img width="752" alt="Screenshot 2024-08-22 at 8 48 13 PM" src="https://github.com/user-attachments/assets/64ceadc6-b734-4351-891f-8a9337c7abaa">

I often find myself trawling through docs/source code like this.

**Reply by kydaly** [reply]
Thank you Daniel! I always wondered how to see the source code and always missed seeing that link. Thank you again for making this free course I have really been enjoying it! I am completely self taught with Python and really appreciate you putting such a thorough intro course up for free.

**Comment by arpit2167407** [comment]
Check the Dataset Class: Look at the FashionMNIST class in the source code of your library (like PyTorch) to see how self.classes and self.class_to_idx are defined.

Print Attributes: Load the dataset and print the attributes:

python
Copy code
from torchvision.datasets import FashionMNIST

dataset = FashionMNIST(root='path/to/data', train=True, download=True)
print(dataset.classes)      # Class names
print(dataset.class_to_idx) # Mapping of class names to indices
Refer to Library Documentation: Check the documentation for the library you're using for additional context.

Community Forums: Look for similar questions on forums like Stack Overflow or GitHub.

Visualize Samples: Use matplotlib to visualize samples from each class for better understanding.
